File: app/src/mock/java/com/amandariu/tagger/Injection.java
package com.amandariu.tagger;

import android.content.Context;
import android.support.annotation.NonNull;

import com.amandariu.tagger.demo.data.FakeTagsRemoteDataSource;
import com.amandariu.tagger.demo.data.source.TagsDataSource;
import com.amandariu.tagger.demo.data.source.TagsRepository;

import static com.google.common.base.Preconditions.checkNotNull;

/**
 * Enables injection of test implementations of the
 * {@link TagsDataSource} at compile time.
 *
 * @author Amanda Riu
 */
public class Injection {

    public static TagsRepository provideTagsRepository(@NonNull Context context) {
        checkNotNull(context);
        return TagsRepository.getInstance(FakeTagsRemoteDataSource.getInstance());
    }
}

File: app/src/mock/java/com/amandariu/tagger/demo/Injection.kt
package com.amandariu.tagger.demo

import android.content.Context
import com.amandariu.tagger.data.FakeTagsRemoteDataSource
import com.amandariu.tagger.demo.data.source.TagsRepository

/**
 * Enables injection of production implementations of the
 * [com.amandariu.tagger.demo.data.source.ITagsDataSource] at compile time.
 *
 * @author Amanda Riu
 */
object Injection {

    fun provideTagsRepository(context: Context): TagsRepository {
        checkNotNull(context)
        return TagsRepository.getInstance(
                FakeTagsRemoteDataSource.getInstance(), FakeTagsRemoteDataSource.getInstance())
    }
}